sympy/solvers/ode.py

```diff
@@ -1064,6 +1079,11 @@ class in it.  Note that a hint may do this anyway if
     if not reduced_eq:
         reduced_eq = eq
 
+    if order == 0:
+        r = _nth_algebraic_match(reduced_eq, func)
+        if r['solutions']:
+            matching_hints['nth_algebraic'] = r
```

Comment:
This code is duplicated below. Make the condition `order >= 0` and remove the code below (although bring the comments up here).

Reply:
Actually thinking about it we always have `order >= 0` don't we? If so we should remove the if.

Reply:
order is always greater than 0. I will remove it.